tests/components/lutron_caseta/test_config_flow.py

```diff
@@ -0,0 +1,122 @@
+        self.is_currently_connected = False
+
+
+async def test_bridge_import_flow(hass):
+    """Test a bridge entry gets created and set up during the import flow."""
+
+    entry_mock_data = {
+        CONF_HOST: "1.1.1.1",
+        CONF_KEYFILE: "",
+        CONF_CERTFILE: "",
+        CONF_CA_CERTS: "",
+    }
+
+    with patch(
+        "homeassistant.components.lutron_caseta.async_setup_entry", return_value=True,
+    ) as mock_setup_entry, patch.object(Smartbridge, "create_tls") as create_tls:
+        create_tls.return_value = MockBridge(can_connect=True)
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data=entry_mock_data,
+        )
+
+    assert result["type"] == "create_entry"
+    assert result["title"] == CasetaConfigFlow.ENTRY_DEFAULT_TITLE
+    assert result["data"] == entry_mock_data
+    await hass.async_block_till_done()
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_bridge_cannot_connect(hass):
+    """Test checking for connection and cannot_connect error."""
+
+    entry_mock_data = {
+        CONF_HOST: "not.a.valid.host",
+        CONF_KEYFILE: "",
+        CONF_CERTFILE: "",
+        CONF_CA_CERTS: "",
+    }
+
+    with patch(
+        "homeassistant.components.lutron_caseta.async_setup_entry", return_value=True,
+    ) as mock_setup_entry, patch.object(Smartbridge, "create_tls") as create_tls:
+        create_tls.return_value = MockBridge(can_connect=False)
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data=entry_mock_data,
+        )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == STEP_IMPORT_FAILED
+    assert result["errors"] == {"base": ERROR_CANNOT_CONNECT}
+    # validate setup_entry was not called
+    assert len(mock_setup_entry.mock_calls) == 0
+
+
+async def test_duplicate_bridge_import(hass):
+    """Test that creating a bridge entry with a duplicate host errors."""
+
+    entry_mock_data = {
+        CONF_HOST: "1.1.1.1",
+        CONF_KEYFILE: "",
+        CONF_CERTFILE: "",
+        CONF_CA_CERTS: "",
+    }
+    mock_entry = MockConfigEntry(domain=DOMAIN, data=entry_mock_data)
+    mock_entry.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.components.lutron_caseta.async_setup_entry", return_value=True,
+    ) as mock_setup_entry:
+        # Mock entry added, try initializing flow with duplicate host
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data=entry_mock_data,
+        )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == CasetaConfigFlow.ABORT_REASON_ALREADY_CONFIGURED
+    assert len(mock_setup_entry.mock_calls) == 0
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)